expression/typeinfer_test.go

```diff
@@ -1249,7 +1249,7 @@ func (s *testInferTypeSuite) createTestCase4TimeFuncs() []typeInferTestCase {
 		{"addtime(c_timestamp, c_time_d)", mysql.TypeDatetime, charset.CharsetBin, mysql.BinaryFlag, 26, 4},
 		{"addtime(c_timestamp_d, c_time_d)", mysql.TypeDatetime, charset.CharsetBin, mysql.BinaryFlag, 26, 0},
 		{"addtime(c_time, c_time)", mysql.TypeDuration, charset.CharsetBin, mysql.BinaryFlag, 15, 3},
-		{"addtime(c_time_d, c_time)", mysql.TypeDuration, charset.CharsetBin, mysql.BinaryFlag, 15, 0},
+		{"addtime(c_time_d, c_time)", mysql.TypeDuration, charset.CharsetBin, mysql.BinaryFlag, 15, 3},
```

Comment:
Do the changes in this file be more reasonable or more compatible?

Reply:
Yes, in previous we didn't infer decimal and always be zero, but in MySQL takes that.

```
mysql> select version();
+---------------------------+
| version()                 |
+---------------------------+
| 5.7.22-0ubuntu18.04.1-log |
+---------------------------+
1 row in set (0.00 sec)

mysql> select addtime(c_time_d, c_time) from t;
Field   1:  `addtime(c_time_d, c_time)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       TIME
Collation:  binary (63)
Length:     14
Max_length: 0
Decimals:   3
Flags:      BINARY 
0 rows in set (0.00 sec)
```